src/outbound.rs

```diff
@@ -263,6 +263,66 @@ impl Outbound {
         })
     }
 
+    pub(crate) fn drain(
+        &self,
+        sids: Vec<usize>,
+        pong_rx: crossbeam_channel::Receiver<bool>,
+    ) -> io::Result<()> {
+        // take lock, which will be held for the duration
+        // of this operation
+        let mut writer = self.writer.lock();
+
+        if writer.is_disconnected() {
+            return Err(Error::new(
+                ErrorKind::NotConnected,
+                "The client is not currently connected to a server",
+            ));
+        }
+
+        if writer.is_closed() {
+            return Err(Error::new(
+                ErrorKind::Other,
+                "the connection is permanently closed",
+            ));
+        }
+
+        // send unsubs
+        for sid in sids {
+            write!(writer, "UNSUB {}\r\n", sid)?;
+        }
+
+        // send PING
+        writer.write_all(b"PING\r\n")?;
+
+        // Flush in place on pings.
+        writer.flush()?;
+
```

Comment:
Have to double check Go client but I believe after this point we want to disallow publish() calls.

Reply:
that is implicit in the writer lock being held